misk/src/test/kotlin/misk/web/actions/UpstreamResourceInterceptorTest.kt

```diff
@@ -43,7 +42,74 @@ internal class UpstreamResourceInterceptorTest {
   }
 
   @Test
-  internal fun requestForwardedToUpstreamServerIfPathMatches() {
+  internal fun mappingFailsIfLocalWithoutLeadingSlash() {
+    var failed = false
+
+    try {
+      mappingTestMapping = UpstreamResourceInterceptor.Mapping(
+          "local/prefix/",
+          upstreamServer.url("/"))
+    } catch (e: IllegalArgumentException) {
+      failed = true
+    }
+
+    assertThat(failed).isTrue()
+  }
+
+  @Test
+  internal fun mappingFailsIfLocalWithoutTrailingSlash() {
+    var failed = false
+
+    try {
+      mappingTestMapping = UpstreamResourceInterceptor.Mapping(
+          "/local/prefix",
+          upstreamServer.url("/"))
+    } catch (e: IllegalArgumentException) {
+      failed = true
+    }
+
+    assertThat(failed).isTrue()
+  }
+
+  @Test
+  internal fun mappingFailsIfUpstreamHasPathSegments() {
+    var failed = false
+
+    try {
+      mappingTestMapping = UpstreamResourceInterceptor.Mapping(
+          "/local/prefix/",
+          upstreamServer.url("/upstream/prefix/"))
+    } catch (e: IllegalArgumentException) {
+      failed = true
+    }
+
+    assertThat(failed).isTrue()
+  }
+
+  @Test
+  internal fun requestForwardedToUpstreamServerIfPathMatchesWithTrailingSlash() {
+    upstreamServer.enqueue(MockResponse()
+        .setResponseCode(404)
```

Comment:
nit: 404 is an awkward choice here, because I think that’s also the code we’ll use if it isn’t intercepted?

Reply:
Yeah we'll change that, already had issues where tests had false positives/negatives because of that.